encoder.go

```diff
@@ -0,0 +1,35 @@
+package jwt
+
+import "io"
+
+// Base64Encoding represents an object that can encode and decode base64. A
+// common example is [encoding/base64.Encoding].
+type Base64Encoding interface {
+	EncodeToString(src []byte) string
+	DecodeString(s string) ([]byte, error)
+}
+
+type StrictFunc[T Base64Encoding] func() T
+
+type Stricter[T Base64Encoding] interface {
```

Comment:
This at least needs a comment if we want to keep It exported, it could probably also be private I guess